examples/siro_sandbox/server/interprocess_record.py

```diff
@@ -0,0 +1,55 @@
+import queue
+
+from .multiprocessing_config import Queue, Semaphore
+
+
+class InterprocessRecord:
+    def __init__(self, max_steps_ahead):
+        self.keyframe_queue = Queue()
+        self.client_state_queue = Queue()
+        self.step_semaphore = Semaphore(max_steps_ahead)
+
+    def send_keyframe_to_networking_thread(self, keyframe):
+        # Acquire the semaphore to ensure the simulation doesn't advance too far ahead
+        self.step_semaphore.acquire()
```

Comment:
The use of this semaphore means the HITL server will "freeze" when not connected to a client. In the future, we can improve this behavior by allowing "anytime connection" of a client. The limitation currently is the incremental nature of gfx-replay keyframes (you need all of them).